src/Magnum/ImGuiIntegration/Integration.cpp

```diff
@@ -0,0 +1,264 @@
+                GL::BufferUsage::StreamDraw);
+
+        for(int c = 0; c < cmdList->CmdBuffer.Size; ++c) {
+            const ImDrawCmd* pcmd = &cmdList->CmdBuffer[c];
+
+            auto userTexture = static_cast<GL::Texture2D*>(pcmd->TextureId);
+            _shader.bindTexture(userTexture ? *userTexture : _texture);
+
+            GL::Renderer::setScissor({
+                    {int(pcmd->ClipRect.x), fbHeight - int(pcmd->ClipRect.w)},
+                    {int(pcmd->ClipRect.z), fbHeight - int(pcmd->ClipRect.y)}});
+
+            _mesh.setCount(pcmd->ElemCount);
+            _mesh.setIndexBuffer(_indexBuffer, indexBufferOffset*sizeof(ImDrawIdx),
+                    sizeof(ImDrawIdx) == 2
+                    ? GL::MeshIndexType::UnsignedShort
+                    : GL::MeshIndexType::UnsignedInt);
+
+            indexBufferOffset += pcmd->ElemCount;
+
+            _mesh.draw(_shader);
+        }
+    }
+}
+
+namespace Implementation {
+
+ImGuiShader::ImGuiShader() {
+    #ifdef MAGNUM_IMGUIINTEGRATION_BUILD_STATIC
+    /* Import resources on static build, if not already */
+    if(!Utility::Resource::hasGroup("MagnumImGuiIntegrationShaders"))
+        importShaderResources();
+    #endif
+    Utility::Resource rs("MagnumImGuiIntegrationShaders");
+
+    #ifndef MAGNUM_TARGET_GLES
+    const GL::Version version = GL::Context::current().supportedVersion({GL::Version::GL330, GL::Version::GL310, GL::Version::GL300, GL::Version::GL210});
+    #else
+    const GL::Version version = GL::Context::current().supportedVersion({GL::Version::GLES300, GL::Version::GLES200});
+    #endif
+
+    GL::Shader vert{version, GL::Shader::Type::Vertex};
+    GL::Shader frag{version, GL::Shader::Type::Fragment};
+
+    #ifndef MAGNUM_TARGET_GLES
+    /* Enable explicit attribute locations in shader if they are supported,
+       otherwise bind locations directly */
+    if(GL::Context::current().isExtensionSupported<GL::Extensions::ARB::explicit_attrib_location>(version)) {
+        vert.addSource({"#define EXPLICIT_ATTRIB_LOCATION\n"});
+    } else {
+        bindAttributeLocation(Position::Location, "position");
+        bindAttributeLocation(TextureCoordinates::Location, "textureCoords");
+        bindAttributeLocation(Color::Location, "color");
+    }
+
+    /* Enable explicit uniform locations if they are supported */
+    if(GL::Context::current().isExtensionSupported<GL::Extensions::ARB::explicit_uniform_location>(version)) {
+        vert.addSource({"#define EXPLICIT_UNIFORM_LOCATION\n"});
+    }
+    #endif
+
+    vert.addSource(rs.get("ImGui.vert"));
+    frag.addSource(rs.get("ImGui.frag"));
+
+    CORRADE_INTERNAL_ASSERT_OUTPUT(GL::Shader::compile({vert, frag}));
+
+    attachShaders({vert, frag});
+
+    CORRADE_INTERNAL_ASSERT_OUTPUT(link());
+
+    _projMatrixUniform = uniformLocation("projectionMatrix");
+
+    setUniform(uniformLocation("colorTexture"), TextureLayer);
```

Comment:
I have a feeling that the two lines above are also wrapped in some `if()` for the builtin shaders. I'll look into this.

Reply:
So after looking into this, what do you say? (I am not sure whether I understand what this is about, btw...)

Reply:
Oh! I'll look into this during the merge, I meant.